File: frontend/src/types/cexProxyBinding.ts
export interface CexProxyBinding {
    bindingId?: number;
    cexName: string;
    proxyId: number;
    proxyName?: string;
    proxyType?: string;
    serverHost?: string;
    serverPort?: number;
    status: 'active' | 'inactive';
    description?: string;
    createdTime?: string;
    updatedTime?: string;
}

export interface CreateCexProxyBindingRequest {
    cexName: string;
    proxyId: number;
    status: 'active' | 'inactive';
    description?: string;
}

export interface UpdateCexProxyBindingRequest {
    cexName: string;
    proxyId: number;
    status: 'active' | 'inactive';
    description?: string;
}

File: backend/src/main/java/com/crypto/trade/entity/CexProxyBinding.java
package com.crypto.trade.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "t_cex_proxy_bindings")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CexProxyBinding {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "binding_id")
    private Long bindingId;

    @Column(name = "cex_name", nullable = false, unique = true, length = 50)
    private String cexName;

    @Column(name = "proxy_id", nullable = false)
    private Long proxyId;

    @Column(name = "status", nullable = false, length = 20)
    private String status = "active";

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "created_time", nullable = false)
    private LocalDateTime createdTime = LocalDateTime.now();

    @Column(name = "updated_time", nullable = false)
    private LocalDateTime updatedTime = LocalDateTime.now();

}
